### PR TITLE
finished week 3 assessment

### PR DESCRIPTION
**Pull request by oagomezz**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ x] Coding Challenges
- [x ] Interview Questions
- [x ] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,19 +13,65 @@
...
 
 // b) Create the function that makes the test pass.
-
+  const fibReturn = (n) => {
+    if (n===2)
+    {
+      return [1, 1];
+    }
+    else
+    {
```
Comment (TsunamiCode5) [inline code-challenges.test.js:39]: Great work!

File: code-challenges.test.js [hunk 2]
```diff
@@ -51,10 +97,32 @@ const studyMinutesWeek2 = {
...
+}
+console.log(smallToBig(studyMinutesWeek1))
+console.log(smallToBig(studyMinutesWeek2))
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
-
+// Function: Create a function that takes in the values of the array and returns the first number plus the second number and so on. 
+// Input: an array of numbers that are to be added accumulatively and added to a secondary array. 
+// Output: an array of numbers with accumulative values. 
+// Method: declare a function.
+// Create a new array to take in the accumulative value of the first value.
```
Comment (TsunamiCode5) [inline code-challenges.test.js:112]: This is really good pseudo code!

File: code-challenges.test.js [hunk 3]
```diff
@@ -65,3 +133,22 @@ const accountTransactions3 = []
...
 
 // b) Create the function that makes the test pass.
+const addToMe = (array) => {
+  let x = array[0]
+  let newArray = []
+  for(i = 0; i < array.length; i++){
+    if(i === 0){
+      let sum = x
+    newArray.push(sum)
+    } else {
```
Comment (TsunamiCode5) [inline code-challenges.test.js:143]: Nice!

File: interview-questions.md [hunk 4]
```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX can be thought of to be a combination of HTML and Javascript, it is utilized in react apps. One syntax difference between HTML and JSX is that in order to access the onclick event in HTML it is written out as onclick whereas to be used in JSX the 'c' in onClick has to be capitalized. JSX can also access javascript functions when being used the only requirement is that {} must be ulitized when calling on the item for example in the use of props in a display component. When they are called from a nested component is it required that the variable be enclosed in {} otherwise the webpage will through up a fault.
```
Comment (TsunamiCode5) [inline interview-questions.md:11]: I love that you included an example because it shows the you definitely have a good grasp on the terms.

**Review by TsunamiCode5 — APPROVED** [review]
Oscar, you did amazing! I would like to see more pseudocode written in as much details as you provided in #3. It has vocabulary, broken down and easy to follow steps, and includes all of the need to know. You are doing a really good job with your interview answers. I like to see that you are giving examples. That is a great way to show an interviewer that you have a solid understanding of the concept and you can even show them how or when it is used. This assessment comprehension level is at Advanced Beginner. A little more detail into your pseudocode and possible refactors in some of your code will definitely bump you into Competent comprehension. Keep up the great work, Oscar!

File: interview-questions.md (hunk 4 above)
Comment (oagomezz) [reply interview-questions.md:11]: Thank you Nicole!!!

**Pull request merged**